python/src/aiconfig/default_parsers/azure.py

```diff
@@ -0,0 +1,40 @@
+from aiconfig.default_parsers.openai import DefaultOpenAIParser
+from aiconfig.util.config_utils import get_api_key_from_environment
+
+from openai import AzureOpenAI
+
+
+class AzureParser(DefaultOpenAIParser):
+    def __init__(self, deployment: str):
+        super().__init__(model_id = deployment)
+
+        # Deployment is a custom name that user defines in Azure portal. 
+        self.deployment = deployment 
+
+    async def deserialize(self, *args, **kwargs):
+        # Logic doesn't depend on input, pass it forward to the openai model parser deserialize 
+        openai_deserialized_params = await super().deserialize(*args, **kwargs)
+
+        # Should this just be the "model name" from config? 
+        # If so, Then the "model name" will have to be deployment name and be consistent 
+        # Which would mean we don't need to have this custom deserialization
+        openai_deserialized_params["model"] = self.deployment 
```

Comment:
Hmm this is tricky. So essentially in azure portal I need to deploy gpt4 and gpt3.5 separately, right? So I'd have to register those model parsers with different deployments and model id keys, right?

Reply:
yep exactly.